src/bindings.rs

```diff
@@ -328,25 +471,14 @@ pub enum nng_pipe_ev {
     NNG_PIPE_EV_REM_POST = 2,
 }
 pub type nng_pipe_cb = ::core::option::Option<
-    unsafe extern "C" fn(
-        arg1: nng_pipe,
-        arg2: ::std::os::raw::c_int,
-        arg3: *mut ::std::os::raw::c_void,
-    ),
+    unsafe extern "C" fn(arg1: nng_pipe, arg2: nng_pipe_ev, arg3: *mut ::core::ffi::c_void),
```

Comment:
This is a source of potential UB that the Rust wrappers can't guard against. C does not enforce that the `nng_pipe_ev` will be one of the allowed values and it is UB in Rust to not be a variant. It may be best to change the Bindgen settings to not use actual enums for the NNG enum types.

Reply:
To be clear, I think having C enums as argument is good (Rust will enforce valid variants) but anywhere an enum is received is immediately potentially UB. See [the Rust language reference](https://doc.rust-lang.org/reference/type-layout.html#reprc-enums).

Reply:
Sure, let's switch them.